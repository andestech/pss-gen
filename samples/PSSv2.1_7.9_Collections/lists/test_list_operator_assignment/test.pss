/*!
@weakgroup      test
@{
@file
@author         Luther Lee
@data           2023/10/25
@brief          Test list assignment operator `=`.
@version
Portable Test and Stimulus Standard Version 2.1 <br>
> Section 7.9 : Collections <br>
> Section 7.9.3 : Lists <br>


---------------------------------------------------------------------------------------------------

### Test Command
@code{.unparsed}
make gentarget
@endcode
*/
component pss_top {

    enum eSTR2NUM {
        ZERO, ONE, TWO
    };

    struct sSTR2NUM {
        string stringVal = "NULL";
        int intVal = 0;
    };

    action root_a {
        list<bit [4] > nibbleList ;
        list<int     > intList    ;
        list<bool    > boolList   ;
        list<eSTR2NUM> enumList   ;
        list<string  > stringList ;
        list<float32 > float32List;
        list<float64 > float64List;
        // list<chandle > chandleList;
        list<sSTR2NUM> structList ;

        sSTR2NUM dummyStruct;

        exec post_solve {
            nibbleList  = {4'b0010, 4'b0011};
            intList     = {2, 3};
            // boolList    = {true, false};
            // enumList    = {TWO, ONE};
            stringList  = {"A", "B"};
            // float32List = {2.2, 2.3};
            // float64List = {3.3.,3.4};
            // structList  = {dummyStruct, dummyStruct};
        }

        exec body ASM =
"""
// test_list_operator_assignment
""";
    }
}

/**@}*/
