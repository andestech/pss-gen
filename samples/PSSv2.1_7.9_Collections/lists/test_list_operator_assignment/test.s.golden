
// test_list_operator_assignment

